core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -438,6 +449,18 @@ public synchronized void reset() throws IOException {
       count = mark;
     }
 
+    @Override
+    public void close() throws IOException {
+      super.close();
+    }
+
+    private void reportCount() {
+      if (count > maxCount) {
+        statsTraceContext.messageReceived(count - maxCount);
```

Comment:
Currently it's doing for uncompressed size, but soon Census will be updated to record both, and I will update my PR.

I only consider being able to record message-reading events is a nice-to-have feature. If it cannot be done accurately, I will keep the current implementation, and rename the methods to something like `wireBytesReceived()` and `uncompressedBytesConsumed()`.